pkgs/applications/editors/emacs/generic.nix

```diff
@@ -46,10 +46,10 @@ assert withXwidgets -> withGTK3 && webkitgtk != null;
 let emacs = stdenv.mkDerivation (lib.optionalAttrs nativeComp {
   NATIVE_FULL_AOT = "1";
   LIBRARY_PATH = "${lib.getLib stdenv.cc.libc}/lib";
-} // lib.optionalAttrs stdenv.isDarwin {
+} // lib.optionalAttrs (stdenv.isDarwin && !stdenv.isAarch64) {
```

Comment:
This CFLAG was added in #41181, but I can't find the exact reason why this was added. It is it something we still need or can we drop it on all platforms? This controls the availability macros, and since we build against the SDK corresponding to our minimum version, this flag should have no effect.

If we keep it, it should probably follow the SDK version on all platforms, like in #124571.

Reply:
Good catch, I build it with `nix-build -E 'with import <nixpkgs> { localSystem = "x86_64-darwin"; }; (import ./default.nix { localSystem = "x86_64-darwin"; }).emacs' -o result` and the binary (.App bundle) ran perfectly fine. Now removed for both platforms.